### PR TITLE
MGMT-694 Download image should use ssh key from parameters instead of…

### PR DESCRIPTION
**Pull request by oshercc**
Download image should use ssh key from parameters instead of

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -199,21 +199,21 @@ func (b *bareMetalInventory) createImageJob(cluster *models.Cluster, name string
 	}
 }
 
-func (b *bareMetalInventory) formatIgnitionFile(cluster *models.Cluster) string {
-	return fmt.Sprintf(ignitionConfigFormat, b.getUserSshKey(cluster), b.AgentDockerImg, b.InventoryURL,
+func (b *bareMetalInventory) formatIgnitionFile(cluster *models.Cluster, params inventory.DownloadClusterISOParams) string {
```
Comment (filanov) [inline internal/bminventory/inventory.go:202]: maybe you should replace the sprintf to `html/template`

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -199,21 +199,36 @@ func (b *bareMetalInventory) createImageJob(cluster *models.Cluster, name string
...
+	var ignitionParams = map[string]string{
+		"userSshKey":     b.getUserSshKey(cluster, params),
+		"AgentDockerImg": b.AgentDockerImg,
+		"InventoryURL":   b.InventoryURL,
+		"InventoryPort":  b.InventoryPort,
+		"clusterId":      cluster.ID.String(),
+	}
+	tmpl, err := template.New("ignitionConfig").Parse(ignitionConfigFormat)
+	if err != nil {
+		panic(err)
```
Comment (filanov) [inline internal/bminventory/inventory.go:212]: NEVER PANIC UNLESS IT'S IN MAIN.GO

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -199,21 +199,36 @@ func (b *bareMetalInventory) createImageJob(cluster *models.Cluster, name string
...
+	}
+	buf := &bytes.Buffer{}
+	if err := tmpl.Execute(buf, ignitionParams); err != nil {
+		panic(err)
```
Comment (filanov) [inline internal/bminventory/inventory.go:216]: NEVER PANIC UNLESS IT'S IN MAIN.GO

**Comment by oshercc** [comment]
recheck

File: internal/bminventory/inventory.go [hunk 4]
```diff
@@ -297,7 +312,12 @@ func (b *bareMetalInventory) DownloadClusterISO(ctx context.Context, params inve
 	}
 	// max job name is 63 chars
 	jobName := fmt.Sprintf("create-image-%s-%s", cluster.ID.String(), uuid.New().String())[:63]
-	if err := b.job.Create(ctx, b.createImageJob(&cluster, jobName)); err != nil {
+	ignitionConfig, formatErr := b.formatIgnitionFile(&cluster, params)
+	if formatErr != nil {
+		log.WithError(formatErr).Error("failed to format ignition config file")
```
Comment (filanov) [inline internal/bminventory/inventory.go:317]: please add cluster id to the log.

**Comment by tsorya** [comment]
@filanov @avishayt if we are using ssh key from download params it means that there is no point to set key in cluster create.
Lets decide how it must work and what point is to set key in cluster params

**Comment by filanov** [comment]
> @filanov @avishayt if we are using ssh key from download params it means that there is no point to set key in cluster create.
> Lets decide how it must work and what point is to set key in cluster params

There are two ssh keys, in cluster create the ssh key that will be used when operating the cluster.
The key on the image is for debugging the installation. it's a way to connect to the host before it's installed.

**Comment by tsorya** [comment]
@filanov as for me if no ssh key is passed to download api , lets use cluster ssh key

**Comment by filanov** [comment]
@oshercc this request pending input from the UX team please don't push

**Pull request merged**